src/content/blog/sweet-sprint-01.md

```diff
@@ -0,0 +1,84 @@
+---
+title: "Sweet Sprint Update 01"
+description: "Time for a do-over. I rewrite argument and file handling for typing exercise selection."
+startDate: "2024-10-02PDT"
+pubDate: "2024-10-02PDT"
+published: true
+heroImage: "/blog/sweet-sprint-01/cover.png"
+project: "sweet"
+tags:
+  - "go"
+---
+
+In case you didn't know, [sweet](https://github.com/NicksPatties/sweet) is the **S**oft**w**are **E**ngineer's **E**xercise for **T**yping I wrote last year. This is a command line application that's designed to help programmers practice typing. I've been thinking about some new features, and wanted to program in Go some more, so I started working on it again! I'm looking forward to getting this in the hands of programmers and getting some feedback.
+
+In the past two weeks, I made some changes to the project to improve my developer experience, but most importantly, I made improvements to selecting different exercises. Allow me to share what I've done!
+
+# An introduction
+
+Since this may be the first time you've heard of `sweet`, I'll share some quick details about how it works.
+
+Running the command opens a random exercise to practice. Type in the exercise, and once done, your stats appear on the screen, and the program ends.
+
+![Default behavior of sweet](@assets/blog/sweet-sprint-01/original-behavior.gif)
+
+## Why `sweet`?
+
+Why make this application anyway? Aside from wanting to learn a new language and writing applications for the command line, I wanted to start a touch typing exercise *as quickly as possible* once I got into my terminal. I didn't want to open a web browser to get my reps in.
+
+If you're interested in practicing your typing, I hope you give `sweet` a try, and that it helps you improve!
+
+# Fixes
+
+## Improve argument handling
+
+Previously, I attempted to roll my own solution for handling sub commands and arguments. Writing help messages for each command, and distinguishing between subcommands and flags quickly got out of hand, resulting in incorrect behavior.
+
+To that end, I decided to hand that responsibility to the [Cobra package](https://github.com/spf13/cobra). This simplified my development by allowing me to focus on subcommands and their individual behaviors instead of argument parsing.
+
+# Updates
+
+To that end, I created some new commands, and improved the touch typing game with some new features!
+
+## New commands: `about` and `version`
+
+These commands are useful for displaying information about the program (In truth, these commands were used as a test to understand how Cobra worked). They share some details about `sweet` and its version number, respectively.
+
+## Better exercise inputs
+
+Previously, `sweet` would read files from an exercise folder in a configuration directory, located in `$HOME/.sweet/exercises`. If the file was not located in that directory, then that exercise would not be added in the rotation. Selecting an individual file to practice was also very cumbersome.
```

Comment:
Previously, `sweet` would ~~read files from an exercise folder in a configuration directory, located~~ grab an exercise from the `$HOME/.sweet/exercises` directory. However, if there was a specific file you'd like to practice that was not included in that directory, selecting it would not work.

Reply:
Point is, edit this line to say that adding files to the exercises directory was cumbersome, and that you could not select exercises by pathname.